pkg/collector/corechecks/oracle-dbm/connection_handling.go

```diff
@@ -97,6 +97,8 @@ func (c *Check) Connect() (*sqlx.DB, error) {
 		c.tags = append(c.tags, fmt.Sprintf("host:%s", c.dbHostname), fmt.Sprintf("oracle_version:%s", c.dbVersion))
 	}
 
+	c.logPrompt = fmt.Sprintf("%s:%s> ", c.cdbName, c.dbHostname)
```

Comment:
This is an interesting feature. I wonder if, instead of setting a logPrompt that we then use and add add to all logs statements, we could create a wrapper logger that would handle this. It's probably an over simplification but maybe something like:

```
type promptLogger struct {
    log.Logger
    prompt string
}

// TODO implement the logging methods and check the prompt to add the prefix

// TODO: use the log wrapper instead when logging
```

Reply:
It's worth noting that the DD log.* functions  show the line which produced the log entry, for example:

`2023-09-15 07:42:39 CEST | CORE | TRACE | (pkg/collector/corechecks/oracle-dbm/activity.go:544 in SampleSession) | Activity payload {"timestamp":1694756559386,"host":"nenad-host-tags","ddsource":"oracle","dbm_type":"activity","ddagentversion":"7.49.0-devel","ddtags":["service:service-haven-seaware","env:development","dbms:oracle","ddagentversion:7.49.0-devel","dbm:true","port:1521","server:nenad-noveljic-cdb-19-ee-3.c7ug0vvtkhqv.us-east-1.rds.amazonaws.com","service:orcl","cdb:rdscdb","host:nenad-host-tags","oracle_version:19.19.0.0.0","hosting_type:RDS"],"collection_interval":1}`

In the above log entry, we immediately see that we have to focus on the line `544` in `activity.go`.

With a log wrapper we would lose this feature, because the code line number that would be printed would always be the log wrapper. It looks like a trade off between a nicer code and troubleshooting agility in production. I'm leaning toward the latter.

What do you think?

Reply:
That's a good point. I think we should preserve line numbers. I'm not sure how that works so I can't suggest a way that would achieve both. Let's just keep it like what you have now.

Reply:
> I'm not sure how that works so I can't suggest a way that would achieve both. 
We could take the call stack in the logger wrapper and print the second line (the first element on the stack is the logger wrapper, and the second is the line that called the wrapper). On the downside, the output would be cluttered, as we would get both lines printed, as the log library would print the logger line.